x-pack/legacy/plugins/reporting/types.d.ts

```diff
@@ -274,16 +226,12 @@ export interface ESQueueInstance {
 
 export type CreateJobFactory<CreateJobFnType> = (
   reporting: ReportingCore,
-  server: ServerFacade,
-  elasticsearch: ElasticsearchServiceSetup,
   logger: LevelLogger
 ) => CreateJobFnType;
 export type ExecuteJobFactory<ExecuteJobFnType> = (
   reporting: ReportingCore,
-  server: ServerFacade,
-  elasticsearch: ElasticsearchServiceSetup,
   logger: LevelLogger
-) => Promise<ExecuteJobFnType>;
+) => ExecuteJobFnType;
```

Comment:
Here's a type change that's perhaps not relevant for this PR. Will keep an eye on this and the CI stability of this PR